src/plugins.js

```diff
@@ -30,62 +30,62 @@ export function setCsrfCookies(request, reply) {
  * function calls `server.state` for both cookie names before registering the
  * plugin.
  *
- * @param {String} secret the 'salt' for encoding the CSRF tokens
- * @return {Object} the { register, options } object for a `server.register` call.
+ * @param {Object} config the environment configuration object
+ * @return {Object} the { register } object for a `server.register` call.
  */
-export function getCsrfPlugin(secret) {
+export function getCsrfPlugin(config) {
 	const register = (server, options, next) => {
 		const cookieOptions = {
 			path: '/',
-			isSecure: process.env.NODE_ENV === 'production',
+			isSecure: options.isProd,
```

Comment:
I was trying that before --- `server.app` was not set at all, so I passed it back in as options. I can see if it's set in `server.settings.app`, but I'm now a little confused about when `server.app` is available (I'm assuming it's not because the top level `server()` call has completed yet), and when we should be using `settings.app` v. `app` within our code.